src/ClientRuntime/Microsoft.Rest.ClientRuntime.E2E.Tests/ScenarioTests/E2ETestBase.cs

```diff
@@ -0,0 +1,793 @@
+
+                return _networkClient;
+            }
+        }
+
+        protected StorageAccount CreateStorageAccount(ResourceGroup resGroup, string storageAccountName)
+        {
+            StorageAccount storageAccountOutput = null;
+            string rgName = resGroup.Name;
+
+            try
+            {
+                var storageAccountList = StorageClient.StorageAccounts.ListByResourceGroup(rgName);
+                if(storageAccountList.Any())
+                {
+                    storageAccountOutput = storageAccountList.Where((sa) => sa.Name.Equals(storageAccountName, StringComparison.OrdinalIgnoreCase)).FirstOrDefault();
+                }
+            }
+            catch { }
+
+            try
+            {
+                if(storageAccountOutput == null)
+                {
+                    var stoInput = new StorageAccountCreateParameters
+                    {
+                        Location = DEFAULT_LOCATION,
+                        AccountType = AccountType.StandardGRS
+                    };
+
+                    storageAccountOutput = StorageClient.StorageAccounts.Create(rgName,
+                        storageAccountName, stoInput);
+                    bool created = false;
+                    while (!created)
+                    {
+                        System.Threading.Thread.Sleep(TimeSpan.FromSeconds(10));
+                        var stos = StorageClient.StorageAccounts.ListByResourceGroup(rgName);
+                        created =
+                            stos.Any(
+                                t =>
+                                    StringComparer.OrdinalIgnoreCase.Equals(t.Name, storageAccountName));
+                    }
+
+                    storageAccountOutput = StorageClient.StorageAccounts.GetProperties(rgName, storageAccountName);
+                }
+
+                return storageAccountOutput;
+            }
+            catch
+            {
+                //ResourceClient.ResourceGroups.Delete(rgName);
+                throw;
+            }
+        }
+
+        protected ResourceGroup CreateResourceGroup(string rgName)
+        {
+            ResourceGroup resourceGroup = null;
+            try
+            {
+                resourceGroup = ResourceClient.ResourceGroups.Get(rgName);
+                //ResourceClient.ResourceGroups.Delete(rgName);
+            }
+            catch { }
+
+            try
+            {
+                if (resourceGroup == null)
+                {
+                    resourceGroup = ResourceClient.ResourceGroups.CreateOrUpdate(
+                                 rgName,
+                                 new ResourceGroup
+                                {
+                                    Location = DEFAULT_LOCATION,
+                                    Tags = new Dictionary<string, string>() { { rgName, DateTime.UtcNow.ToString("u") } }
+                                });
+                }
+            }
+            catch
```

Comment:
@markcowl will remove, lot of code is remnant of trial and error in this really long running test.